tools/wptrunner/wptrunner/manifestinclude.py

```diff
@@ -81,74 +81,52 @@ def _get_components(self, url):
-            last, query = last.rsplit("?", 1)
-
-        maybe_path = os.path.join(rest, last)
-        paths = glob.glob(maybe_path)
-
-        if paths:
-            urls = []
-            for path in paths:
-                for manifest, data in test_manifests.items():
-                    found = False
-                    rel_path = os.path.relpath(path, data["tests_path"])
-                    iterator = manifest.iterpath if os.path.isfile(path) else manifest.iterdir
-                    for test in iterator(rel_path):
-                        if not hasattr(test, "url"):
-                            continue
-                        url = test.url
-                        if query or fragment:
-                            parsed = urlparse(url)
-                            if ((query and query != parsed.query) or
-                                (fragment and fragment != parsed.fragment)):
-                                continue
-                        urls.append(url)
-                        found = True
-                    if found:
-                        break
-        else:
-            urls = [url]
-
-        assert direction in ("include", "exclude")
-
-        for url in urls:
-            components = self._get_components(url)
-
-            node = self
-            while components:
-                component = components.pop()
-                if component not in node.child_map:
-                    new_node = IncludeManifest(DataNode(component))
-                    node.append(new_node)
-                    new_node.set("skip", node.get("skip", {}))
-
-                node = node.child_map[component]
-
-            skip = False if direction == "include" else True
-            node.set("skip", str(skip))
-
-    def add_include(self, test_manifests, url_prefix):
-        """Add a rule indicating that tests under a url path
-        should be included in test runs
-
-        :param url_prefix: The url prefix to include
-        """
-        return self._add_rule(test_manifests, url_prefix, "include")
-
-    def add_exclude(self, test_manifests, url_prefix):
-        """Add a rule indicating that tests under a url path
-        should be excluded from test runs
 
-        :param url_prefix: The url prefix to exclude
-        """
-        return self._add_rule(test_manifests, url_prefix, "exclude")
+def resolve_pattern(test_manifests, pattern):
+    maybe_path = os.path.join(os.path.abspath(os.curdir), pattern)
+    rest, last = os.path.split(maybe_path)
+    fragment = query = None
+    if "#" in last:
+        last, fragment = last.rsplit("#", 1)
+    if "?" in last:
+        last, query = last.rsplit("?", 1)
+
+    maybe_path = os.path.join(rest, last)
+    paths = glob.glob(maybe_path)
+
+    urls = []
+    if paths:
+        for path in paths:
+            for manifest, data in test_manifests.items():
+                found = False
+                rel_path = os.path.relpath(path, data["tests_path"])
```

Comment:
Apparently not, since `testloader.ManifestLoader.load()` still returns `Dict[manifest.Manifest, Dict[str, str]]`:

https://github.com/web-platform-tests/wpt/blob/aeaebad53eab2288fb9d26ba9f2888361ead74aa/tools/wptrunner/wptrunner/testloader.py#L331-L334

which is plumbed through to `testloader.TestLoader.manifests` and consumed here in `resolve_pattern()`.

(It seems worthwhile to add `TestRoot.url_base` and pass the same `TestRoot` everywhere, but that's for another PR.)